### ISSUE TITLE
0.9.3 BETA: Playing Artist Plays Albums Alphabetically instead of Chronologically

### ISSUE THREAD

**Issue by 9p4** [jmshrv/finamp#609]
When pressing the "play" button on an artist, the expected behaviour is to play the albums chronologically. Instead the albums are queued up alphabetically.

**Comment by Chaphasilor** [comment]
Any ideas how this should be handled? Do you want to change the default behavior? Have an option in settings to change between the two behaviors? Have an additional playback option? Have a way to sort artist albums and always play in that order?

**Comment by 9p4** [comment]
I think that the previous behaviour ought to be kept: play the albums chronologically. That is what I expected (and what happens in other music-playing apps as well)

**Comment by hoggatt** [comment]
I agree queuing in order should be the default. It would also be nice to be able to customize the artist page a bit. For example, maybe I just want to see the albums and not the top songs. Also, I usually want the albums ordered from newest to oldest, so allowing this to be a setting would be nice too.

**Comment by Tiefseetauchner** [comment]
The issue is the way Jellyfin sorts the songs grouped by album. The real question here is whether we want to locally sort the already sorted songs, or whether we should investigate Jellyfins options further. I don't know what Jellyfin can do, but I know that performance is an issue for me on my Fairphone 3 (given, an old and even back then slow phone)

**Comment by Chaphasilor** [comment]
I'd like to rely on Jellyfin for sorting as much as possible. Sadly, the web version doesn't sort chronologically either (probably due to similar problems), and Jellyfin-Vue only queues a single track.

I guess the only way to properly achieve this is by loading all albums, sorted chronologically, and then loading the tracks for each album afterwards

**Comment by Tiefseetauchner** [comment]
Good luck! If I may provide a bit of ideas:

Load the albums chronologically and then grab each albums songs until you reached a threshold (~~last I remember, which tbf it's been a while, this is also used on Genres~~ scrap that it isn't anymore) so to not load a thousand songs into the queue at once.

I'm literally walking through a forest, so my memory may be off, but I believe I put in a limit in the query, I'd use that. Maybe even grab the song count of the album in the initial query and fire the queries in a future.all or whatever it was called, if it's not possible or easy to append to the queue as the album lists come in. If it is, do just add them sequentially, probably better than a hundred simultaneous queries.

Anyways, just wanted to distract myself from this weird thing called *nature* and *trees* (ew), probably do not have to tell ya all that.
